packages/platform/src/HttpClient.ts

```diff
@@ -549,6 +549,22 @@ export const tap: {
   ): HttpClient.With<E | E2, R | R2>
 } = internal.tap
 
+/**
+ * Performs an additional effect after an unsuccessful request.
+ *
+ * @since 1.0.0
+ * @category mapping & sequencing
+ */
+export const tapError: {
+  <_, E, E2, R2>(
+    f: (e: E) => Effect.Effect<_, E2, R2>
```

Comment:
```suggestion
    f: (e: NoInfer<E>) => Effect.Effect<_, E2, R2>
```

Should probably use NoInfer here to prevent overriding the error type.